File: filesize.rb
require 'gets'

print 'Enter file size (MB): '
the_size = gets.chomp

fail 'bad file size' unless the_size =~ /^\d+$/

file_size = 0
string    = 'abcdefghijklmnopqrstuvwxyz0123456789'

File.open(the_size + 'MB', 'w') do |f|
  while file_size < the_size.to_i * 1048576 # bytes in 1MB
    f.print string
    file_size += string.size
  end
end

File: filesize.cpp
#include <iostream>
#include <fstream>
#include <string>

using namespace std;

double fileSize = 0;
double theSize;
string fileName;
string chars = "abcdefghijklmnopqrstuvwxyz0123456789";

int generateDataInMB(string fileName, double theSize) {
  if (theSize <= 0) {
    cout << "Enter a valid number! \n";
    generateDataInMB;
  } else {
    ofstream file;
    file.open(fileName);
    while (fileSize < (theSize * 1048576)) { //bytes in 1MB
      file << chars;
      fileSize += chars.length(); }
  file.close();
  cout << "Done!" << endl;
  }
}

int main() {
  cout << "Enter file size in (MB): ";
  cin  >> theSize;

  fileName = to_string(theSize) + "MB.txt"; // Adds decimal point to fileName

  generateDataInMB(fileName, theSize);
}